flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/BackendRestorerProcedure.java

```diff
@@ -50,7 +49,7 @@
  * @param <S> type of the supplied snapshots from which the backend restores.
  */
 public class BackendRestorerProcedure<
-	T extends Closeable & Disposable & Snapshotable<?, Collection<S>>,
+	T extends Closeable & Disposable & SnapshotStrategy<?>,
```

Comment:
We can remove the `& SnapshotStrategy<?>` part here, becaues we don't invoke anymore methods from that interface.